### ISSUE TITLE
Total requests per second in graph is vastly different from actual stats

### ISSUE THREAD

**Issue by ErikSchlyter** [locustio/locust#2300]
### Describe the bug
I was monitoring a distributed load test with 24 attached workers, and the chart in the web UI showed the current load to be around 150 request/second. When collecting the logs afterwards to print them in my own gnuplot, I noticed my graphs showing around 450-500 request/second. I inspected the generated report, and the report itself states the aggregated average request/second to be 421.3, which is vastly different from what the chart is showing.

### Expected behavior
The chart in the web UI should show a good approximation of the amount of request/second.

### Actual behavior
![total_requests_per_second_1676991856](https://user-images.githubusercontent.com/692418/220405187-f68468fb-d82e-4dd4-abdb-445332df528d.png)
![Screenshot 2023-02-21 171944](https://user-images.githubusercontent.com/692418/220405220-dbbff61c-ef88-4475-adad-524fcbfa3870.png)

### Steps to reproduce
Conduct a distributed load test from 24 attached workers, running in individual Docker containers throughout western Sweden. The actual test code was sending HTTPS POST-request towards 2 separate web services.

### Environment

- OS: Docker contained based on python:3.11.1-alpine3.17
- Python version: 3.11.1
- Locust version: 2.14.2
- Locust command line that you ran: locust -f locustfiles --worker --master-host my-ssh-tunnel-proxy --master-port 5557
- Locust file contents (anonymized if necessary):
```python
import locust


class SoapUser(locust.FastHttpUser):
    abstract = True
    key_cert_file = None # needs to be set in sub-class

    def __init__(self, environment):
        super().__init__(environment=environment)

        def _create_default_context():
            context = gevent.ssl.create_default_context()
            context.check_hostname = False
            context.verify_mode = gevent.ssl.CERT_NONE
            if not self.key_cert_file is None:
                context.load_cert_chain(certfile=self.key_cert_file)
            return context

        self.client.client.clientpool.client_args['ssl_context_factory'] = _create_default_context

    def request(self, url, action, data):
        headers = {
            "ContentType": "application/xml",
            "SOAPAction": action
        }
        return self.client.request('POST', url=url, headers=headers, data=data, catch_response=True)

class xyz(SoapUser):
    wait_time = locust.constant(1)

    key_cert_file = '/mnt/secret/secret.pem' 

    def request(self, path, action, payload):
        """Send request and verify that we get resultCode OK in return"""
        with super().request(path, action, payload) as response:
            if response.status_code == 200 and not "<resultCode>OK</resultCode>" in response.text:
                response.failure("Did not receive resultCode OK")

    @locust.task
    def get_xyz_1(self):
        self.request( path='/xyz/1', action='"xyz"', payload=f""" <soapenv:Envelope xyzxyzxyz>anonymized</soapenv:Envelope> """)

    @locust.task
    def get_xyz_2(self):
        self.request( path='/xyz/2', action='"xyz"', payload=f""" <soapenv:Envelope xyzxyz>anonymized</soapenv:Envelope> """)
```

Please let me know if there's any other detail I can contribute. I have anonymized the actual XML content in the HTTP post, but this should show the gist of it.

**Comment by cyberw** [comment]
> Conduct a distributed load test from 24 attached workers, running in individual Docker containers throughout western Sweden. The actual test code was sending HTTPS POST-request towards 2 separate web services.

That may be hard to reproduce :)

Have you checked the logs? The only reason I can think of is that the workers cannot (for some reason) report back to the master..

If there is nothing clear in the logs you’ll need to narrow it down further some other way, as the issue you are encountering is unlikely to be ”locust always makes an error counting number of requests” (or everyone else would be complaining about this too :)

**Comment by ErikSchlyter** [comment]
Yeah, I get that :)

I've checked the master logs, but I can't see anything weird during that particular execution. There was some test-runs before the given graph with some disconnecting and reconnecting workers just to verify that everything was up to speed, but nothing out of the ordinary.

The fact that the report contains an aggregated 421.3 req/sec and simultaneously shows this graph leads me to believe master actually _did_ receive the stats, but something goes wrong somewhere. The total number of requests also matches my external report (the workers contain additional code to write all requests to disk, which I collect afterwards for more precise stats).

![locust_report](https://user-images.githubusercontent.com/692418/220602861-a7ad733a-517d-4b10-b838-0b16032a700f.png)

I know this would be super hard to debug and I don't expect much, but I just wanted the report this behavior to have a dialog running. My current theory is that these different Docker containers run on different host OS (Windows/Linux), which might affect their comprehension of time. I suppose the `_log_time_of_request` would be totally off if each Worker reported a different value from `time.time()`, right?

I will continue to investigate on my side. I have some more tests to run.

**Comment by ErikSchlyter** [comment]
I have executed some tests during the day, and it seems some of the Docker containers were running on hosts without NTP enabled. The time was off with 3-4 minutes between the machines, which hints to my above mentioned theory that the wrong value from `time.time()` will create a mess when the stats are aggregated into the chart. Once I set the time, I couldn't reproduce the problem anymore.

I assume the current request per second is computed from some kind of window/timespan based on the Master node's current time, which would explain why a lot of requests were omitted even though they are included in the stats.

I'm not sure how to fix it, beyond perhaps implementing some kind of warning if the workers' reference of time differs too much from master's, but if feels a bit nit-picky. I'll close this issue until further notice.

Thanks!
